src/main/java/nl/novi/serviceconnect/services/ServiceRequestService.java

```diff
@@ -3,55 +3,66 @@
 import nl.novi.serviceconnect.exceptions.RecordNotFoundException;
+import nl.novi.serviceconnect.exceptions.UsernameNotFoundException;
 import nl.novi.serviceconnect.helpper.Mapper;
 import nl.novi.serviceconnect.helpper.Helpers;
 import nl.novi.serviceconnect.models.ServiceRequest;
-import nl.novi.serviceconnect.models.Transaction;
+import nl.novi.serviceconnect.models.User;
 import nl.novi.serviceconnect.repository.ServiceRepository;
 import nl.novi.serviceconnect.repository.ServiceRequestRepository;
+import nl.novi.serviceconnect.repository.UserRepository;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 
 @Service
 public class ServiceRequestService implements IServiceRequest{
-    private final ServiceRequestRepository repository;
+    private final ServiceRequestRepository serviceRequestRepository;
     private final ServiceRepository serviceRepository;
+    private final UserRepository userRepository;
 
-    public ServiceRequestService(ServiceRequestRepository repository, ServiceRepository serviceRepository) {
-        this.repository = repository;
+    public ServiceRequestService(ServiceRequestRepository repository, ServiceRepository serviceRepository, UserRepository userRepository) {
+        this.serviceRequestRepository = repository;
         this.serviceRepository = serviceRepository;
+        this.userRepository = userRepository;
     }
 
     @Override
-    public ServiceRequestOutputDto createServiceRequest(ServiceRequestInputDto serviceRequestInputDto) {
-        ServiceRequest serviceRequestResult =  repository.save(Mapper.fromDtoToServiceRequest(serviceRequestInputDto));
+    public ServiceRequestOutputDto createServiceRequest(ServiceRequestInputDto serviceRequestInputDto, String username) {
+        ServiceRequest serviceRequestResult =  Mapper.fromDtoToServiceRequest(serviceRequestInputDto);
 
-        nl.novi.serviceconnect.models.Service request = serviceRepository.findById(serviceRequestInputDto.getService().getId())
+        nl.novi.serviceconnect.models.Service service = serviceRepository.findById(serviceRequestInputDto.getService().getId())
                 .orElseThrow(() -> new RecordNotFoundException("Service not found with id: " + serviceRequestInputDto.getService().getId()));
-        serviceRequestResult.setService(request);
+        serviceRequestResult.setService(service);
 
-        Transaction transaction = new Transaction();
-        transaction.setTransactionDate(new Date());
-        transaction.setPayed(false);
+        if (username != null && !username.isEmpty()) {
+            User user = userRepository.findById(username).orElseThrow(() -> new RecordNotFoundException("User not found with username: " + username));
+            serviceRequestResult.setUser(user);
+        } else {
+            throw new UsernameNotFoundException("Username is empty or null");
+        }
```

Comment:
Dit kun je aan het begin van de methode zetten, omdat het een Unhappy flow is. Hoe eerder hij daarop reageert hoe efficiënter je applicatie werkt.